### DISCUSSION TITLE
[Feature] Exclude face recognition before birth

### DISCUSSION THREAD

**Discussion by DrPeterVC** [immich-app/immich#4743] [Feature Request]
### The feature

It is great to be able specify the date of birth to assist with face recognition.  But the algorithm still seems to attempt face recognition even before birth.  So I have a photo of me as a baby flagged as my youngest grandson aged -57.  So please can we have "ignore before birth" feature for face recognition.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by titro** [comment]
Don't unterstand the purpose? If there is a negative birth it could only the case if y photo is assigend to a wrong person?
So there should not be a "ignore before birth" rather than a "show recognitions with problems/warnings" section to solve or correct these recognitions?

**Reply by DrPeterVC** [reply]
There are basically two possible solutions:
1.  When face recognition is running don't include people whose birth date is before the create date of the photo.  In which case a rerun of facial recognition would remove the wrong assignment(s).
Or your proposal:
2. Include a feature to show problems with face recognition and allow manual correction (re scanning would potentially just put the wrong assignment again).
IMHO The first option would be a simpler feature to implement.

**Comment by DrPeterVC** [comment]
I have double checked and redone all the face recognition with the latest version 1.84 and it is still doing facial recognition on children before they were born!  
I am contemplating developing a database script to go through and remove faces from assets that are before birth.  
IMHO this should be a simple feature to implement at the recognition stage.

**Reply by bo0tzz** [reply]
> IMHO this should be a simple feature to implement at the recognition stage.

A PR would be welcome!

**Reply by DrPeterVC** [reply]
Hi - I am an enthusiastic user of immich and have some programming experience.  I have explored the structure of the database and can see how it is put together.  But I have not looked at the code at all.  If someone can point me in the right direction I could certainly look.  It must be a lot easier to introduce this feature at the code level than at the database level.

**Reply by bo0tzz** [reply]
Here is the primary code that handles face recognition: https://github.com/immich-app/immich/blob/2a75f884d9ab63f355bd60c44a76a6e8a2c05c43/server/src/domain/person/person.service.ts#L381

**Reply by DrPeterVC** [reply]
Thanks I will study the best way to handle this.

**Comment by abhinav96** [comment]
Thanks for pointing to the core piece of code @bo0tzz .
I have submitted a PR with @DrPeterVC 's first suggested solution.
Could anyone please take a look and suggest if it can be merged? https://github.com/immich-app/immich/pull/16918

Thanks

**Comment by aviv926** [comment]
This feature was added via PR #16918

**Reply by Hamish1066** [reply]
How do I use this feature? I can’t see anything in mobile.
